lib/chef/resource/systemd_service.rb

```diff
@@ -0,0 +1,49 @@
+#
+# Author:: Davide Cavalca (<dcavalca@fb.com>)
+# Copyright:: Copyright (c) 2015 Facebook
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/resource/service'
+require 'chef/provider/service/systemd'
+
+class Chef
+  class Resource
+    class SystemdService < Chef::Resource::Service
+      provides :service, os: "linux" do |node|
+        Chef::Platform::ServiceHelpers.service_resource_providers.include?(:systemd)
+      end
```

Comment:
Yeah, but then you wind up hardcoding the `system` resource to the systemd service resource on the DSL for any system which has systemd and you lose the ability to lazy evaluate to a different provider later -- or worse you might wind up picking a systemd resource and pairing it up with a non-systemd resource at some later point.  Its not a big deal now because currently systemd assimilates distros like the borg, but i wouldn't necessarily assume that will always be the case forever.  And I'd rather put up with the service resource sprouting inaccurate validations than remove that flexibility.